src/VirtoCommerce.OpenIdConnectModule.Core/Models/OidcOptions.cs

```diff
@@ -0,0 +1,134 @@
+using System.Collections.Generic;
+
+namespace VirtoCommerce.OpenIdConnectModule.Core.Models
+{
+    public class OidcOptions
+    {
+
+        /// <summary>
+        /// Determines whether the user authentication via OpenId Connect is enabled.
+        /// </summary>
+        public bool Enabled { get; set; }
+
+        /// <summary>
+        /// Sets AuthenticationType value for OpenId Connect authentication provider.
+        /// </summary>
+        public string AuthenticationType { get; set; }
+
+        /// <summary>
+        /// Sets human-readable caption for OpenId Connect authentication provider. It is visible on sign-in page.
+        /// </summary>
+        public string AuthenticationCaption { get; set; }
+
+        /// <summary>
+        /// Application ID of the VirtoCommerce platform application registered in OpenId Connect.
+        /// </summary>
+        public string ApplicationId { get; set; }
+
+        /// <summary>
+        /// URL of the OpenId Connect endpoint used for authentication.
+        /// </summary>
+        public string Authority { get; set; }
+
+        /// <summary>
+        /// Default user type for users created by OpenId Connect accounts.
+        /// </summary>
+        public string DefaultUserType { get; set; }
+
+        /// <summary>
+        /// Default user roles for users created by OpenId Connect accounts.
+        /// </summary>
+        public string[] DefaultUserRoles { get; set; }
+
+        /// <summary>
+        /// Allow creating new user when a user authenticates via IDP for the first time
+        /// </summary>
+        public bool AllowCreateNewUser { get; set; } = true;
+
+        /// <summary>
+        /// Display dedicated login form or not
+        /// </summary>
+        public bool HasLoginForm { get; set; } = true;
+
+        /// <summary>
+        ///  Gets or sets the discovery endpoint for obtaining metadata
+        /// </summary>
+        public string MetadataAddress { get; set; }
+
+        /// <summary>
+        /// Client Id 
+        /// </summary>
+        public string ClientId { get; set; }
+
+        /// <summary>
+        /// Client Id 
+        /// </summary>
+        public string ClientSecret { get; set; }
+
+        /// <summary>
+        /// Client Id 
+        /// </summary>
+        public string CallbackPath { get; set; } = "/signin-oidc";
```

Comment:
It's better to use a different callback URLs for each external provider to avoid errors when unpacking the message
`/signin-oidc` is already used by default by AzureAD
https://josef.codes/unable-to-unprotect-the-message-state-asp-net-core/